app/controllers/topics_controller.rb

```diff
@@ -6,15 +6,30 @@ def index
 
   def show
     @topic = policy_scope(Topic).find(params[:id])
-    @signs = policy_scope(@topic.signs)
+
+    search_results ||= TopicSignService.new(search:, relation: policy_scope(Sign), topic: @topic).process
```

Comment:
nit: there's no point conditionally assigning here

```suggestion
    search_results = TopicSignService.new(search:, relation: policy_scope(Sign), topic: @topic).process
```